### DISCUSSION TITLE
reading/writing non-default file tags?

### DISCUSSION THREAD

**Discussion by presentday-presenttime** [beetbox/beets#5358] [Q&A]
I am using some existing tags, that aren't natively supported by beets (from picard, Mp3tag, or wherever). I want to be able to read/write them using beets. is there a way to do this?

using the `modify`  command afaik doesn't actually write to the file unless it's one of the already supported tags.

I've tried using the `inline` plugin with `mutagen` but it's slow, and I couldn't figure out how to support writing.

**Comment by Serene-Arc** [comment]
Beets doesn't really support arbitrary metadata fields out of the box. You'd need a plugin for that. What specific tags do you mean?

**Reply by presentday-presenttime** [reply]
rating and series are ones I use often that I handle outside of beets. at one point I was looking at mood & some I can't think of right now.

Mostly I was just wondering if there was a more straightforward way than having to write a plugin.

**Reply by Serene-Arc** [reply]
AFAIK there isn't, there would need to be fields defined somewhere, which is usually through a plugin. I think that's the usual way to define flexible fields, which would be what you'd use.

@snejus are you aware of a way to do this without a plugin?

**Reply by snejus** [reply]
You're right @Serene-Arc, we don't have a plugin that allows writing custom media fields. @presentday-presenttime, see [this GitHub code search](https://github.com/search?q=self.add_media_field+language%3Apython&type=code) for the relevant function that should be used to set custom media fields. There are a couple of plugins that do this, so they may serve as a starting point for you if you'd like to go ahead and code this
